cmd/root.go

```diff
@@ -132,6 +131,10 @@ has completed`,
 			cancel()
 		}()
 
+		if !hasEphemeralContainersResource(client) {
+			klog.Fatal(fmt.Errorf("api server does not have pods/ephemeralcontainers resource"))
```

Comment:
Do we want to use fatal here? I'm wondering if there are instances where Ephemeral containers are not available from the API server, we can use an alternative approach? Perhaps the same strategy that we used prior to the implementation of ephemeral containers?

I am likely missing context here. Just my 2 cents

Reply:
Nono, it's something I've been thinking about. It's definitely a possibility, however, the previous implementation relied on the sidecars/containers we are targeting to contain the ability to send some type of signal to kill the main process. It's a solution, though one I don't find as elegant due to the increased attack surface of the containers for such a simple requirement. Ending pods shouldn't require dependencies in the Container itself.

The reason I added this check early on was to fail-fast and essentially notify operators that this solution wouldn't work ASAP, since, depending on the cluster and it's usage, there may not be a sidecar to terminate for days or months.

I was thinking that if someone really requires the `exec kill` functionality they can use v1 of the solution. 

Eventually, I would like to polish this up into an operator!